src/mapping-v3-core.ts

```diff
@@ -508,10 +711,64 @@ function createProject(
   project.pricePerTokenInWei = pricePerTokenInWei;
   project.projectId = projectId;
   project.scriptCount = scriptCount;
-  project.updatedAt = timestamp;
   project.useHashString = useHashString;
   project.useIpfs = false;
 
+  // populate the project's financial information
+  // available on all cores, populated to non-zero value for new projects on v3.2+ cores
+  project.royaltyPercentage = contract.projectIdToSecondaryMarketRoyaltyPercentage(
+    projectId
+  );
+  // provider secondary financials are modeled differently on v3.0 and v3.1 vs. v3.2+, handle both cases
+  const coreVersion = contractEntity.version;
+  // version should never be null on a V3 contract
+  if (!coreVersion) {
+    log.error(
+      "V3 Contract entity for contract {} does not have a version set and has caused a subgraph error.",
+      [contract._address.toHexString()]
+    );
+    return null;
+  }
+  const isPreV3_2 = getIsPreV3_2(coreVersion as string);
```

Comment:
Does this work fine now? I feel like I remember always having to use `changetype` for casting in assemblyscript

Reply:
I think still have to use changetype if changing from one type to another - this was required to let AS compiler know `coreVersion` wasn't null for some reason, even though there is already an early return immediately before this statement 🤷 honestly a little confusing to me, but without, 
```
FAILURE 1 compile error(s)
ERROR TS2322: Type '~lib/string/String | null' is not assignable to type '~lib/string/String'.

   const isPreV3_2 = getIsPreV3_2(coreVersion);
```